### ISSUE TITLE
Feature request: Rolling ols takes a time window when the by column is a datetime

### ISSUE THREAD

**Issue by Ge0rges** [azmyrajab/polars_ols#37]
It would be convenient if, when `x` is a `datetime` datatype that `rolling_ols` can take a time window such as `1m`, instead of a. number of points.

**Comment by azmyrajab** [comment]
Hi @Ge0rges 

does this pattern work for you 
https://github.com/azmyrajab/polars_ols/issues/23

That would fallback to polars rolling syntax which is the most flexible option and it would run the regressions in parallel over windows, it should still be fast

**Comment by azmyrajab** [comment]
Yes it is less elegant to break lazy chaining of an expression I suppose. Sorry I haven’t tested what I’m saying now, but could you try something like:

`pls.ols(y, x, …).rolling(index_column=“my_date”, period=“30m”).overy(my_group)`

if that works, that should do exactly what you want it to
